src/Avalonia.X11/X11CursorFactory.cs

```diff
@@ -52,13 +55,16 @@ internal partial class X11CursorFactory : ICursorFactory
 
         [GenerateEnumValueList]
         private static partial CursorFontShape[] GetAllCursorShapes();
-        
+
         public X11CursorFactory(IntPtr display)
         {
             _display = display;
             _nullCursor = GetNullCursor(display);
-            _cursors = GetAllCursorShapes()
-                .ToDictionary(id => id, id => XLib.XCreateFontCursor(_display, id));
+
+            var cursorShapes = GetAllCursorShapes();
+            _cursors = new Dictionary<CursorFontShape, IntPtr>(cursorShapes.Length);
+            foreach (var shape in cursorShapes)
+                _cursors[shape] = XLib.XCreateFontCursor(_display, shape);
```

Comment:
Is `XCreateFontCursor` a bottleneck? If so, I think this call could be done lazily on `GetCursor`.

Reply:
I think that's a viable strategy aswell.

Without initializing the cursors at all, startup is around 50ms.
Hence why I noted the savings were less than what I expected, and there might have been some contention.

It might actually be better than the current solution; as the user is unlikely to immediately jump in and use all cursor styles. Let me make a quick commit to test this.

Reply:
After some testing, I've it seems that lazy initialization for cursors is probably the better approach.
It gets a startup time of `40-51ms` on my machine, so around twice as fast than the baseline.

I've committed bfb2ef20b31e77850aa7e89f1ce51ebbd951df67 which achieves that as the current commit and updated the opening post. The previous PR is still available in the branch history.